table/block_based/block_based_table_reader.cc

```diff
@@ -619,6 +626,33 @@ Status BlockBasedTable::Open(
   if (!s.ok()) {
     return s;
   }
+
+  // With properties loaded, we can set up portable/stable cache keys if
+  // necessary info is available
+  std::string db_session_id = cur_db_session_id;
+  uint64_t file_num = cur_file_num;
+  if (rep->table_properties && !rep->table_properties->db_session_id.empty()) {
+    const auto& uprops = rep->table_properties->user_collected_properties;
+    auto version_iter = uprops.find(ExternalSstFilePropertyNames::kVersion);
+    if (version_iter == uprops.end()) {
+      // Normal (non-external) SST file - can only use embedded db_session_id
+      // with current file number (which should be original file number)
+      if (file_num > 0) {
+        db_session_id = rep->table_properties->db_session_id;
+      }
+    } else {
+      // External (ingested) SST file - should not use current file number
+      // (changed from original), so that same file ingested into different DBs
+      // can share block cache entries. Although they can modify the embedded
+      // global_seqno, that information is not currently cached under these
+      // portable/stable keys.
+      // Note: For now, each external SST file gets its own unique session id.
```

Comment:
Maybe also mention that cc->NewId() will be used instead file_number as the suffix.

Reply:
Oops! My intention was to use a fixed file number as suffix but 0 means "unknown."

I will fix this and add to unit test.

Reply:
Actually, fixing it requires a bit more intervention than I want to throw into this PR after approval.

Reply:
I will take a look after you push the updates.